src/IMangrove.sol

```diff
@@ -8,70 +8,51 @@ pragma solidity >=0.7.0 <0.9.0;
-    address indexed inbound_tkn,
-    uint indexed tickScale,
-    address owner,
-    address spender,
-    uint value
-  );
-  event Credit(address indexed maker, uint amount);
-  event Debit(address indexed maker, uint amount);
-  event Kill();
-  event NewMgv();
-  event OfferFail(bytes32 indexed olKeyHash, uint id, uint takerWants, uint takerGives, uint penalty, bytes32 mgvData);
-  event OfferRetract(bytes32 indexed olKeyHash, uint id, bool deprovision);
-  event OfferSuccess(bytes32 indexed olKeyHash, uint id, uint takerWants, uint takerGives);
-  event OfferWrite(
-    bytes32 indexed olKeyHash, address maker, int logPrice, uint gives, uint gasprice, uint gasreq, uint id
-  );
-  event OrderComplete(
-    bytes32 indexed olKeyHash, address taker, uint takerGot, uint takerGave, uint penalty, uint feePaid
-  );
-  event OrderStart();
-  event PosthookFail(bytes32 indexed olKeyHash, uint offerId, bytes32 posthookData);
-  event SetActive(bytes32 indexed olKeyHash, bool value);
-  event SetDensityFixed(bytes32 indexed olKeyHash, uint value);
-  event SetFee(bytes32 indexed olKeyHash, uint value);
-  event SetGasbase(bytes32 indexed olKeyHash, uint offer_gasbase);
-  event SetGasmax(uint value);
-  event SetGasprice(uint value);
-  event SetGovernance(address value);
-  event SetMonitor(address value);
-  event SetNotify(bool value);
-  event SetUseOracle(bool value);
-
+interface IMangrove is HasMgvEvents {
   function DOMAIN_SEPARATOR() external view returns (bytes32);
 
   function PERMIT_TYPEHASH() external view returns (bytes32);
 
   function withdrawERC20(address tokenAddress, uint value) external;
-  function activate(OLKey memory olKey, uint fee, uint density, uint offer_gasbase) external;
+  function activate(OLKey memory olKey, uint fee, uint densityFixed, uint offer_gasbase) external;
 
-  function allowances(address, address, address, address) external view returns (uint);
+  function allowances(address outbound_tkn, address inbound_tkn, address owner, address spender)
+    external
+    view
+    returns (uint allowance);
 
   function approve(address outbound_tkn, address inbound_tkn, address spender, uint value) external returns (bool);
 
-  function balanceOf(address) external view returns (uint);
+  function balanceOf(address maker) external view returns (uint balance);
 
-  function best(OLKey memory olKey) external view returns (uint);
+  function best(OLKey memory olKey) external view returns (uint offerId);
 
-  function config(OLKey memory olKey) external view returns (MgvStructs.GlobalPacked, MgvStructs.LocalPacked);
+  function config(OLKey memory olKey)
+    external
+    view
+    returns (MgvStructs.GlobalPacked _global, MgvStructs.LocalPacked _local);
 
   function configInfo(OLKey memory olKey)
     external
     view
-    returns (MgvStructs.GlobalUnpacked memory global, MgvStructs.LocalUnpacked memory local);
+    returns (MgvStructs.GlobalUnpacked memory _global, MgvStructs.LocalUnpacked memory _local);
```

Comment:
Again, why the underscores?

Reply:
This is the same signature as for your other comment. So they are just identical :)

"config" was the one with already existing _global.